widgets/blueprints/src/TerraformModal.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useState } from 'react';
+import type { DropdownProps } from 'semantic-ui-react';
+import TerraformModalAccordion from './TerraformModalAccordion';
+import TerraformModalTableAccordion from './TerraformModalTableAccordion';
+import TerraformVariableValueInput from './TerraformVariableValueInput';
+import TerraformActions from './TerraformActions';
+import terraformVersions from './terraformVersions';
+import type { CustomConfigurationComponentProps } from '../../../app/utils/StageAPI';
+import type { Variable, Output } from '../../../backend/routes/Terraform.types';
+
+const t = Stage.Utils.getT('widgets.blueprints.terraformModal');
+
+const { Dropdown } = Stage.Basic;
+
+const terraformVersionOptions = terraformVersions.map(versionOption => ({
+    text: versionOption,
+    value: versionOption
+}));
+terraformVersionOptions[0].text = `${terraformVersionOptions[0].text} (${t('default')})`;
+
+function getDynamicTableDropdown(options: DropdownProps['options']) {
+    return ({ name, onChange, ...rest }: CustomConfigurationComponentProps<string>) => {
+        return (
+            <Dropdown
+                clearable={false}
+                selection
+                options={options}
+                onChange={(event, { value }) => onChange?.(event, { name, value: value as string })}
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...rest}
+            />
+        );
+    };
+}
+
+const dynamicTableFieldStyle = { height: 38 };
+
+const variablesColumns = [
```

Comment:
To avoid similar problems with `id` values in the future we could have:
```suggestion
interface VariablesColumn extends Stage.Types.WidgetConfigurationDefinition {
    id: keyof Variable;
}
const variablesColumns: VariablesColumn[] = [
```
That will allow us to have only valid `id` values allowed. WDYT?

The same comment applies to `outputsColumns`.

Reply:
Smart idea 🙂. I slightly improved it by moving `id` type constraint to `TerraformModalTableAccordionProps`.